examples/cu_consolemon/src/lib.rs

```diff
@@ -0,0 +1,376 @@
+                Constraint::Length(10),
+                Constraint::Length(12),
+                Constraint::Length(12),
+                Constraint::Length(10),
+                Constraint::Length(12),
+                Constraint::Length(13),
+            ],
+        )
+        .header(header)
+        .block(Block::default().borders(Borders::ALL).title(" Latencies "));
+
+        f.render_widget(table, area);
+    }
+
+    fn draw(&self, f: &mut Frame) {
+        let layout = Layout::default()
+            .direction(Direction::Vertical)
+            .constraints(
+                [
+                    Constraint::Length(3), // For the top menu bar
+                    Constraint::Min(0),    // For the main content
+                ]
+                .as_ref(),
+            )
+            .split(f.area());
+
+        let menu = Paragraph::new("   [1] SysInfo  [2] DAG  [3] Latencies  [q] Quit")
+            .style(
+                Style::default()
+                    .fg(Color::Yellow)
+                    .add_modifier(Modifier::ITALIC),
+            )
+            .block(Block::default().borders(Borders::BOTTOM));
+        f.render_widget(menu, layout[0]);
+
+        match self.active_screen {
+            Screen::Neofetch => {
+                let text: Text = format!("\n{}", self.sysinfo).into_text().unwrap();
+                let p = Paragraph::new::<Text>(text).block(
+                    Block::default()
+                        .title(" System Info ")
+                        .borders(Borders::ALL),
+                );
+                f.render_widget(p, layout[1]);
+            }
+            Screen::Dag => {
+                let p = Paragraph::new(dag(&self.config))
+                    .block(Block::default().title(" Tasks DAG ").borders(Borders::ALL));
+                f.render_widget(p, layout[1]);
+            }
+            Screen::Latency => self.draw_latency_table(f, layout[1]),
+        };
+    }
+
+    fn run_app<B: Backend>(&mut self, terminal: &mut Terminal<B>) -> io::Result<()> {
+        loop {
+            terminal.draw(|f| {
+                self.draw(f);
+            })?;
+
+            if event::poll(Duration::from_millis(10))? {
+                if let Event::Key(key) = event::read()? {
+                    match key.code {
+                        KeyCode::Char('1') => self.active_screen = Screen::Neofetch,
+                        KeyCode::Char('2') => self.active_screen = Screen::Dag,
+                        KeyCode::Char('3') => self.active_screen = Screen::Latency,
+                        KeyCode::Char('r') => {
+                            if self.active_screen == Screen::Latency {
+                                self.task_stats.lock().unwrap().reset()
+                            }
+                        }
+                        KeyCode::Char('q') => {
+                            // Basically triggers the clean ctrlc logic of copper.
+                            signal::kill(
```

Comment:
I don't think this is resetting the terminal correctly. When i do `q` it exits but dumps out a bunch of ansi color commands

![Screenshot 2024-10-02 210506](https://github.com/user-attachments/assets/8452f835-49d7-43f7-ba20-9230f93bf6d8)

I have to shut down the terminal to stop it from spitting out those strings endlessly.

I'm running Ubuntu 24.04 using WSL2

Reply:
Ok this threw me into a refactor, there were other shudown issues in the core too :). Now I send an error to the runtime on the process copperlist that makes the runtime shutting down correctly (and there were gremlins in the unified logger that was not closing properly that got fixed too)